GoogleForCreators/web-stories-wp
# Editor: Layer renaming

## Context


Editing layers names

## Summary


Editing of the layers names can be enabled by:
- double clicking on the layers names;
- clicking Rename Layer in the RCM of the element;
- clicking Rename Layer in the RCM of the layers menu;

A name is saved by clicking Enter, clicking outside of the layer button (Blur).
The editing is canceled without saving a new name buy clicking Escape.

## Relevant Technical Choices


layerName property was added to the element object.

## To-do


1. Several components can be extracted from the layers.js file to separate files.
2. Keyboard events handler functions can be extracted to utils so they can be reused.

## User-facing changes


<img width="363" alt="image" src="https://user-images.githubusercontent.com/2496186/168179910-20628cab-aee6-4f33-a383-be81414f3d2f.png">
<img width="371" alt="image" src="https://user-images.githubusercontent.com/2496186/168179991-2c54b668-9daa-4891-9ecd-064a7a6d5be0.png">
<img width="353" alt="image" src="https://user-images.githubusercontent.com/2496186/168180030-0008410f-d146-490a-b6ac-9e205370f6ac.png">


## Testing Instructions






This PR can be tested by following these steps:

Renaming successful scenario:
1. Double click on the Name -> an input appears -> type a new name -> click Enter -> the layer has a new name.
2. Right click on the element -> a menu opens -> click Rename Layer -> an input appears -> type a new name -> click Enter -> the layer has a new name.
3. Right click on a layer button in the Layers menu -> a menu opens -> click Rename Layer -> an input appears -> type a new name -> click Enter -> the layer has a new name.
4. Repeat 1 - 4, but instead of clicking Enter, click outside of the layer button (Blur).

Renaming canceled scenario:
5. Repeat 1 - 4, but click Escape button instead of Enter -> the layer name should not be updated.

Note: when an input field is visible, layer action icons should be hidden.

## Checklist




---



Fixes #9248
